src/IECoreArnold/ShaderNetworkAlgo.cpp

```diff
@@ -317,6 +317,40 @@ ShaderNetworkPtr preprocessedNetwork( const IECoreScene::ShaderNetwork *shaderNe
 	ShaderNetworkPtr result = shaderNetwork->copy();
 	IECoreScene::ShaderNetworkAlgo::convertOSLComponentConnections( result.get() );
 	IECoreArnold::ShaderNetworkAlgo::convertUSDShaders( result.get() );
+
+	/// Convert `quad_light` width and height, if needed.
+	if( result->outputShader()->getName() == "quad_light" )
+	{
+		ShaderNetwork::Parameter outputParameter = result->getOutput();
+
+		ShaderPtr newShader = result->getShader( outputParameter.shader )->copy();
+
+		const auto verticesDataIt = newShader->parameters().find( "vertices" );
+		if( verticesDataIt == newShader->parameters().end() )
+		{
+			float width = 2.f;
+			float height = 2.f;
+
+			if( auto widthData = newShader->parametersData()->member<FloatData>( "width" ) )
+			{
+				width = widthData->readable();
+			}
+			if( auto heightData = newShader->parametersData()->member<FloatData>( "height" ) )
+			{
+				height = heightData->readable();
+			}
+
+			newShader->parameters()["vertices"] = new V3fVectorData( {
+				V3f( width / 2, height / 2, 0 ),
+				V3f( width / 2, -height / 2 , 0 ),
+				V3f( -width / 2, -height / 2, 0 ),
+				V3f( -width / 2, height / 2, 0 )
```

Comment:
One possibility for the sampling change in the test is that our vertex ordering differs from Arnold's default :

```import arnold

arnold.AiBegin()

n = arnold.AiNodeEntryLookUp( "quad_light" )
p = arnold.AiNodeEntryLookUpParameter( n, "vertices" )
d = arnold.AiParamGetDefault( p )
a = d.contents.ARRAY
for i in range( 0, arnold.AiArrayGetNumElements( a ) ) :
	print( i, ":", arnold.AiArrayGetVec( a, i ).x, arnold.AiArrayGetVec( a, i ).y, arnold.AiArrayGetVec( a, i ).z )

arnold.AiEnd()
```

Outputs :

```
0 : -1.0 -1.0 0.0
1 : -1.0 1.0 0.0
2 : 1.0 1.0 0.0
3 : 1.0 -1.0 0.0
```

Reply:
That definitely fixes the test, and I think we need to adhere to that ordering to keep from reorienting everyone's quad light textures when they update to the version this ends up in.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5169/commits/7b28509c04775978ef866e9255b7ce3d92fbb516